contracts/DivviRegistry.sol

```diff
@@ -31,6 +39,21 @@ contract DivviRegistry is
     address indexed rewardsProvider,
     address indexed rewardsConsumer
   );
+  event ReferralRegistered(
+    address user,
+    address indexed rewardsProvider,
+    address indexed rewardsConsumer,
+    uint256 indexed chainId,
+    bytes32 txHash
+  );
+  event ReferralSkipped(
+    address indexed user,
+    address rewardsProvider,
+    address indexed rewardsConsumer,
+    uint256 chainId,
+    bytes32 indexed txHash,
+    string reason
+  );
```

Comment:
Regarding the `ReferralSkipped` event: emitting `string reason` is readable but more gas-intensive, especially within `batchRegisterReferral`.

I recommend replacing it with the `ReferralStatus status` enum value for better gas efficiency. Off-chain systems can still map the status code to a description.

Let me know your thoughts.

Reply:
yes it's the readability that made me lean towards a string, but you're right that anyone who is interested can map the enum to a string reason if they need it. gas efficiency isn't a good tradeoff then, i updated to use the enum